### DISCUSSION TITLE
how can I implement fft-generator on genesys2??

### DISCUSSION THREAD

**Discussion by ttcheng26** [eugene-tarassov/vivado-risc-v#247] [Q&A]
Dear sir,

I have a question about add a fft-generator base on rocket-chip in genesys2, how can i do ?
(since chipyard has a generator called fft-generator, but in this repo didn't have it)

**Comment by eugene-tarassov** [comment]
You need to learn Chisel and SBT and you will know how to add modules from any other Chisel repo including chipyard.

**Reply by cycyucheng1010** [reply]
Dear Sir,

I am currently researching the FFT Generator and attempting to implement it within this repository. I have added some content and settings. What intrigues me is that while I can successfully execute sbt test and make BOARD=genesys2 CONFIG=Rocket64b1fft8 ROCKET_FREQ_MHZ=80 bitstream, I am unable to locate the FFT Generator in any of the results.

here is some of my setting :
[github build.sbt](https://github.com/cycyucheng1010/Vivado-RISCV/blob/master/build.sbt)
[github rocket.scala](https://github.com/cycyucheng1010/Vivado-RISCV/blob/master/src/main/scala/rocket.scala)

**Reply by eugene-tarassov** [reply]
Your changes look good to me. I would expect FFT block to show up somewhere in the generated system-genesys2.v. However I have no experience with chipyard FFT Generator, I'm not sure how exactly it is supposed to work.

**Reply by cycyucheng1010** [reply]
Dear @eugene-tarassov,

Thank you for your response. I have thoroughly checked the system-genesys2.v file and can confirm that there is no module related to the FFT generator or the memory locations I set. This conclusion was also supported by my observations in Vivado.

I would like to seek your expertise on a more advanced topic: I am curious about how you integrate the Rocket SoC with the FPGA board in your project. Currently, the FFT generator can be simulated and verified within Chipyard using Verilator. I hope to leverage this approach to address the issue and make the project more comprehensive and automated.

Best Regards,
Rick Chen
